src/main/plugins/org.dita.pdf2/xsl/fo/root-processing.xsl

```diff
@@ -265,9 +265,21 @@ See the accompanying license.txt file for applicable licenses.
     <xsl:apply-templates select="*" mode="generatePageSequences"/>
   </xsl:template>
   <xsl:template match="*[contains(@class, ' map/topicref ')]" mode="generatePageSequences" priority="0">
-    <xsl:for-each select="key('topic-id', @id)">
-      <xsl:call-template name="processTopicSimple"/>
-    </xsl:for-each>
+    <xsl:choose>
+      <xsl:when test="ancestor::*[contains(@class,' bookmap/frontmatter ')]">
+        <!-- To fit the pattern, this should be in its own match template. But a general match for frontmatter/*

```

Comment:
Add `TODO` prefix to comments that have a description of what the code should do once refactored.